### DISCUSSION TITLE
仕様メモ

### DISCUSSION THREAD

**Discussion by hikuohiku** [TERA1103/Settings-Beautify#1] [General]
# 機能要件
- **Settings.jsonのソート機能**
	- 拡張機能ごとに分ける
- **インラインコメントをいい感じにつける**
	- 拡張機能ごとのメインコメント
		- Descriptionなど書けるようにしたい
	- 設定キーごとのインラインコメント
		- キーの上の行をそのキーのコメントとしてソートするようにする
- **特定の設定項目に対するお気に入り機能**

**Comment by hikuohiku** [comment]
とりあえずコマンド単位で機能を作成していく

**Reply by hikuohiku** [reply]
実装するコマンド群↓
- Settings.json（ユーザースペース）をフォーマット
- Settings.json（ワークスペース）をフォーマット
- Settings.json（両方）をフォーマット

**Comment by TERA1103** [comment]
機能案

・ 拡張機能ごとのトグルボタン
・ 編集日時を表示

**Comment by hikuohiku** [comment]
settings.jsonを拡張機能側から直接編集してしまうのはリスクが高くセキュリティ的に怪しい（かつ，VS Code APIの機能だけでは実現できなさそう）ので，ユーザーにSettings.jsonの内容をコピー/ペーストで入力してもらい，拡張機能側からもユーザーにクリップボードベースで出力する方式にする．
